### ISSUE TITLE
Saving Annotation Failed in Internet Explorer 11

### ISSUE THREAD

**Issue by robertknight** [hypothesis/client#506]
_Transferred from https://github.com/hypothesis/h/issues/4610_

![ie11 issue](https://user-images.githubusercontent.com/26881125/28540815-095eb218-708d-11e7-883b-2d74a0f5dee5.png)
### Steps to reproduce
1. Browse a page where Hypothesis is available using Internet Explorer 11
2. Log into Hypothesis
3. Attempt to post an annotation

### Expected behaviour
The annotation would save as expected.

### Actual behaviour
An error is displayed - "Saving annotation failed 400: document.link.0: 'href' is a required property"

### Browser/system information
Internet Explorer 11.0.9600.17420 on Windows 7
Internet Explorer 11.1000.15014.0 on Windows 10

### Additional details
Annotation will successfully save when logging with the same account on a different browser within the same OS.

**Comment by robertknight** [comment]
I was able to reproduce this in one of Microsoft's Windows 10/Edge VirtualBox VMs trying to annotate https://web.hypothes.is/blog/dois/ and http://robertknight.github.io/posts/react-europe-2016/ using IE 11.483.15063.0. I didn't experience the problem when using Via to annotate example.org (https://via.hypothes.is/https://example.org).

This error is visible in the console:

```
TypeError: Unable to get property 'startsWith' of undefined or null reference
   at Anonymous function (https://cdn.hypothes.is/hypothesis/1.32.0/build/scripts/app.bundle.js?f82c09:377:576)
   at searchUrisForFrame (https://cdn.hypothes.is/hypothesis/1.32.0/build/scripts/app.bundle.js?f82c09:377:405)
   at Anonymous function (https://cdn.hypothes.is/hypothesis/1.32.0/build/scripts/app.bundle.js?f82c09:377:725)
   at searchUris (https://cdn.hypothes.is/hypothesis/1.32.0/build/scripts/app.bundle.js?f82c09:377:688)
   at n[r] (https://cdn.hypothes.is/hypothesis/1.32.0/build/scripts/app.bundle.js?f82c09:392:341)
   at m (https://cdn.hypothes.is/hypothesis/1.32.0/build/scripts/app.bundle.js?f82c09:251:1349)
   at Anonymous function (https://cdn.hypothes.is/hypothesis/1.32.0/build/scripts/app.bundle.js?f82c09:251:2315)
   at d.prototype.$broadcast (https://cdn.hypothes.is/hypothesis/1.32.0/build/scripts/angular.bundle.js?0c54ba:19:12582)
   at a (https://cdn.hypothes.is/hypothesis/1.32.0/build/scripts/app.bundle.js?f82c09:341:398)
   at focusGroup (https://cdn.hypothes.is/hypothesis/1.32.0/build/scripts/app.bundle.js?f82c09:209:522)
   at Anonymous function (https://cdn.hypothes.is/hypothesis/1.32.0/build/scripts/angular.bundle.js?0c54ba:21:8237)
   at t (https://cdn.hypothes.is/hypothesis/1.32.0/build/scripts/angular.bundle.js?0c54ba:19:1076)
   at i (https://cdn.hypothes.is/hypothesis/1.32.0/build/scripts/angular.bundle.js?0c54ba:21:18674)
   at d.prototype.$eval (https://cdn.hypothes.is/hypothesis/1.32.0/build/scripts/angular.bundle.js?0c54ba:19:11190)
   at d.prototype.$apply (https://cdn.hypothes.is/hypothesis/1.32.0/build/scripts/angular.bundle.js?0c54ba:19:11422)
   at Anonymous function (https://cdn.hypothes.is/hypothesis/1.32.0/build/scripts/angular.bundle.js?0c54ba:21:18724)
   at We (https://cdn.hypothes.is/hypothesis/1.32.0/build/scripts/angular.bundle.js?0c54ba:17:15497)
   at n (https://cdn.hypothes.is/hypothesis/1.32.0/build/scripts/angular.bundle.js?0c54ba:17:15446)
   at n (https://cdn.hypothes.is/hypothesis/1.32.0/build/scripts/raven.bundle.js?3d4f55:9:6308) undefined
```

**Comment by robertknight** [comment]
This is the body of the annotation that the client attempted to send. Note the empty entry in the `links` array:

```json
{
  "created": "2017-07-25T07:08:54.024Z",
  "updated": "2017-07-25T07:08:54.024Z",
  "document": {
    "highwire": {},
    "dc": {},
    "facebook": {
      "title": [
        "Federating Annotations Using Digital Object Identifiers (DOIs)"
      ],
      "type": [
        "article"
      ],
      "url": [
        "https://web.hypothes.is/blog/dois/"
      ],
      "site_name": [
        "Hypothesis"
      ],
      "description": [
        "Scientific journals come and go, but the scientific record is permanent, and its annotation layer should be too. New Hypothesis support for DOIs (digital object identifiers) helps ensure a robust connection between articles and annotations. Let's explore how that works.\n\n\nFirst, here's a magic trick you might not realize Hypothesis has up its sleeve. Consider"
      ],
      "image": [
        "https://d242fdlp0qlcia.cloudfront.net/uploads/2017/06/22182746/doi-search.jpg"
      ]
    },
    "eprints": {},
    "prism": {},
    "twitter": {},
    "favicon": "https://d242fdlp0qlcia.cloudfront.net/uploads/2017/04/16172247/hypothesis-favicon16x16.png",
    "title": "Federating Annotations Using Digital Object Identifiers (DOIs)",
    "link": [
      {},
      {
        "href": "https://web.hypothes.is/blog/dois/",
        "rel": "canonical",
        "type": ""
      },
      {
        "href": "https://web.hypothes.is/?p=8642",
        "rel": "shortlink",
        "type": ""
      },
      {
        "href": "https://web.hypothes.is/wp-json/oembed/1.0/embed?url=https%3A%2F%2Fweb.hypothes.is%2Fblog%2Fdois%2F",
        "rel": "alternate",
        "type": "application/json+oembed"
      },
      {
        "href": "https://web.hypothes.is/wp-json/oembed/1.0/embed?url=https%3A%2F%2Fweb.hypothes.is%2Fblog%2Fdois%2F&format=xml",
        "rel": "alternate",
        "type": "text/xml+oembed"
      }
    ]
  },
  "uri": "https://web.hypothes.is/blog/dois/",
  "target": [
    {
      "source": "https://web.hypothes.is/blog/dois/",
      "selector": [
        {
          "type": "RangeSelector",
          "startContainer": "/div[2]/div[4]/div[1]/div[1]/article[1]/div[2]/p[1]",
          "startOffset": 1,
          "endContainer": "/div[2]/div[4]/div[1]/div[1]/article[1]/div[2]/p[1]",
          "endOffset": 21
        },
        {
          "type": "TextPositionSelector",
          "start": 787,
          "end": 807
        },
        {
          "type": "TextQuoteSelector",
          "exact": "Scientific journals ",
          "prefix": "2017|\t\t\t\n\t\t\t\t\t\t\t\t\t\t\n\t\t\t\t\t\t\n\t\t\t\t\n",
          "suffix": "come and go, but the scientific "
        }
      ]
    }
  ],
  "user": "acct:robertknight@hypothes.is",
  "group": "__world__",
  "permissions": {
    "read": [
      "group:__world__"
    ],
    "update": [
      "acct:robertknight@hypothes.is"
    ],
    "delete": [
      "acct:robertknight@hypothes.is"
    ]
  },
  "text": "Test in IE 11",
  "tags": []
}
```

**Comment by robertknight** [comment]
The problem is in the `_getDocumentHref` function introduced as part of the EPUB work: https://github.com/hypothesis/client/blob/e903ebaca38363f150446aaadf2b353e2f0b5fb5/src/annotator/plugin/document.coffee#L193

`baseURI` (which comes from the [document-base-uri](https://github.com/tilgovi/document-base-uri/blob/a81e32b210be68eb1035a5a0fb8c9e388707f9c0/index.js) package) is undefined here because IE does not appear to support either the `Node.baseURI` or `Document.documentURI` properties.

**Comment by robertknight** [comment]
Thank-you for the report. This issue is fixed in v1.32.1 which is rolling out over the next couple of hours.
